src/main/java/ch/njol/skript/bukkitutil/HealthUtils.java

```diff
@@ -112,8 +112,11 @@ public static double getFinalDamage(EntityDamageEvent e) {
 		return e.getFinalDamage() / 2;
 	}
 	
-	public static void setDamage(EntityDamageEvent e, double damage) {
-		e.setDamage(damage * 2);
+	public static void setDamage(EntityDamageEvent event, double damage) {
+		event.setDamage(damage * 2);
+		// Set last damage manually as Bukkit doesn't appear to do that
+		if (event.getEntity() instanceof LivingEntity)
+			((LivingEntity) event.getEntity()).setLastDamage(damage * 2);
```

Comment:
my only concern is if this value changes as the result of a later event. it may be in our best interest to keep referencing getLastDamageCause?

Reply:
Unless I'm misreading something here, isn't that what we want?
if a damage event is called, and changes this value... that's what we want right?
Why would we not want that?
Also lastDamageCause... would that not do the same thing?